### DISCUSSION TITLE
Register User Without company

### DISCUSSION THREAD

**Discussion by mstfkhazaal** [andrewdwallo/filament-companies#58] [Feature requests]
Please add the feature to allow the user to register without a company, and he can add later a company if he wants

**Comment by andrewdwallo** [comment]
Working on it!

**Reply by andrewdwallo** [reply]
Actually, depending on what you mean, it may be a no. If you are talking about company users and not company owners, then I am working on where an invited employee cannot create their own company if the developer decides to do that, but a company owner who signs up will have to have a company.

**Reply by mstfkhazaal** [reply]
I mean that the person who registers does not establish a company with him, but he can establish a company at another time
A person can have a company and be an employee of another company

**Reply by andrewdwallo** [reply]
Currently, a person can have a company and be an employee of another company. So yes.

**Comment by ronaldvaneede** [comment]
> Actually, depending on what you mean, it may be a no. If you are talking about company users and not company owners, then I am working on where an invited employee cannot create their own company if the developer decides to do that, but a company owner who signs up will have to have a company.

Is this already in there? Maybe I missed this but could not find it in the documentation.
If not when do you expect this to be included?

**Comment by bogusbd** [comment]
Here is a good post describing current problems and solutions with Team/ company  Invitations,  maybe will be useful to someone

https://mariogiancini.com/making-laravel-jetstream-team-invitations-better

**Comment by pablozagni** [comment]
If a new user registers, it will create a company. Then, he can invite his co-workers to join the company. So, users accept the invitation an join the company.
I don't see why a user would register and don't create a company.

**Reply by ronaldvaneede** [reply]
If that co-worker does not yet have an account he needs to create one. 
As a result that co-worker also has to create a personal company. That is what I want to avoid.
What we want is that only the person that created the 'main' account can create companies (or tenants if you will), in our case branches.

**Reply by pablozagni** [reply]
Yes, we have to disable team creation to workers.. so no personal companies are created
Let me work a little bit more and I'll show you some code.
I't seems to me that it's a Jetstream related option...

**Reply by pablozagni** [reply]
As I think, if a user REGISTERS, it will create a new company. Then, he can INVITE other co-workers.
If a co-worker ACCEPTS the invitation to join the company, it will register, create the new user, and join the invited comany, with no need to create a new one.
Is it ok ?

**Reply by andrewdwallo** [reply]
That sounds correct.

**Comment by andrewdwallo** [comment]
It is worth noting that if anyone is willing to make a PR for such a feature, where a company owner can invite their employees to join the company (either by creating the account for them or by them creating an account after receiving the invite), and then for those employees to not have to create a company (personal company) after registering.. I am willing to accept a PR for this. I would probably have to create a new major version but that is fine with me.

**Reply by bryangruneberg** [reply]
@andrewdwallo - would you be open to a sponsorship to build this into the package in a well considered and documented way? I think lots of people will need and want this, and seeing it in the package would be really great!

**Reply by andrewdwallo** [reply]
@bryangruneberg yeah possibly. It would depend on the sponsorship.
